src/cryptoadvance/specter/managers/wallet_manager.py

```diff
@@ -348,19 +348,24 @@ def create_wallet(self, name, sigs_required, key_type, keys, devices, **kwargs):
         else:
             raise ("Failed to create new wallet")
 
-    def delete_wallet(
-        self, wallet, bitcoin_datadir=get_default_datadir(), chain="main"
-    ):
+    def delete_wallet(self, wallet, datadir, chain="main"):
         logger.info("Deleting {}".format(wallet.alias))
         wallet_rpc_path = os.path.join(self.rpc_path, wallet.alias)
         self.rpc.unloadwallet(wallet_rpc_path)
         # Try deleting wallet folder
-        if bitcoin_datadir:
+        logger.debug(
+            f"This is the datadir when deleting wallet in wallet_manager: {datadir}"
+        )
+        logger.debug(
+            f"This is the wallet rpc path when deleting wallet in wallet_manager: {wallet_rpc_path}"
+        )
+
+        if datadir:
             if chain != "main":
-                bitcoin_datadir = os.path.join(bitcoin_datadir, chain)
+                datadir = os.path.join(datadir, chain)
             candidates = [
-                os.path.join(bitcoin_datadir, wallet_rpc_path),
-                os.path.join(bitcoin_datadir, "wallets", wallet_rpc_path),
+                os.path.join(datadir, wallet_rpc_path),
+                os.path.join(datadir, "wallets", wallet_rpc_path),
             ]
             for path in candidates:
                 shutil.rmtree(path, ignore_errors=True)
```

Comment:
Please move the deletion logic inside the node. Can't we have a node-method called `delete_wallet` which will do whatever is necessary?!

Reply:
done in https://github.com/cryptoadvance/specter-desktop/pull/1950/commits/5b6280f60473750e578dc3d46e5319d605894984